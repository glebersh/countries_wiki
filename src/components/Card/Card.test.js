import { render, screen } from '@testing-library/react';
import { toBeInTheDocument, toHaveBeenCalledWith } from '@testing-library/jest-dom';
import { Card } from './Card';
import { useNavigate } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

const countryInfo = {
  img: 'https://via.placeholder.com/300.png',
  name: 'Brasil',
  info: [
    {
      title: 'Population',
      description: 214047375
    },
    {
      title: 'Region',
      description: 'Latin Amerika',
    },
    {
      title: 'Capital',
      description: 'BrasÃ­lia ',
    },
  ],
};

describe('Card component', () => {
  it('Card renders', () => {
    render(<Card {...countryInfo} />)

    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByRole('img')).toHaveAttribute('alt', 'Brasil');
    expect(screen.getByRole('img')).toHaveAttribute('src', countryInfo.img);
    expect(screen.getByRole('list')).toBeInTheDocument();
  });

  it('Card snapshot', () => {
    const countryCard = render(<Card {...countryInfo} />);
    expect(countryCard).toMatchSnapshot();
  });

  it('Navigate to DetailsPage', () => {
    const navigate = useNavigate();
    render(<Card {...countryInfo} onClick={() => navigate(`country/${countryInfo.name}`)} />);
    userEvent.click(screen.getByRole('article'));
    expect(mockedUsedNavigate).toHaveBeenCalledWith('country/Brasil');
  });
});
