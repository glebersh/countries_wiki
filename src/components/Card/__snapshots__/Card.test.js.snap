// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="sc-bcXHqe ipQmHr"
      >
        <img
          alt="Brasil"
          class="sc-gswNZR cfWXzP"
          src="https://via.placeholder.com/300.png"
        />
        <div
          class="sc-dkrFOg hXwiuy"
        >
          <h3
            class="sc-hLBbgP krmYgr"
          >
            Brasil
          </h3>
          <ul
            class="sc-eDvSVe juJjhm"
          >
            <li
              class="sc-jSUZER jhitMP"
            >
              <b>
                Population
                :
              </b>
               
              214047375
            </li>
            <li
              class="sc-jSUZER jhitMP"
            >
              <b>
                Region
                :
              </b>
               
              Latin Amerika
            </li>
            <li
              class="sc-jSUZER jhitMP"
            >
              <b>
                Capital
                :
              </b>
               
              Brasília 
            </li>
          </ul>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="sc-bcXHqe ipQmHr"
    >
      <img
        alt="Brasil"
        class="sc-gswNZR cfWXzP"
        src="https://via.placeholder.com/300.png"
      />
      <div
        class="sc-dkrFOg hXwiuy"
      >
        <h3
          class="sc-hLBbgP krmYgr"
        >
          Brasil
        </h3>
        <ul
          class="sc-eDvSVe juJjhm"
        >
          <li
            class="sc-jSUZER jhitMP"
          >
            <b>
              Population
              :
            </b>
             
            214047375
          </li>
          <li
            class="sc-jSUZER jhitMP"
          >
            <b>
              Region
              :
            </b>
             
            Latin Amerika
          </li>
          <li
            class="sc-jSUZER jhitMP"
          >
            <b>
              Capital
              :
            </b>
             
            Brasília 
          </li>
        </ul>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
